src/argilla/__init__.py

```diff
@@ -34,6 +35,14 @@
 except ModuleNotFoundError:
     pass
 
+if _sys.version_info < (3, 8):
+    warnings.warn(
+        message="Your Python runtime version is deprecated and will not be supported in the upcoming release. "
```

Comment:
Just a nit, but I'd just mention that Python 3.7 is not supported

```suggestion
        message="Python 3.7 is coming to its end-of-life and will be no longer supported in the upcoming release of Argilla. "
```